core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -159,6 +169,9 @@
    * Assumes no rule needs more than 2 slots. */
   protected final MutableRel[] slots = new MutableRel[2];
 
+  // FOR TESTING.
+  public static boolean disabled = false;
```

Comment:
Double negative is counter intuitive, can we give it `enabled = true`?

Reply:
Or remove it, or make it private. Someone is going to set that field and cause a problem in production.

Reply:
+1 to remove it.